MaterialLib/MPL/Properties/CapillaryPressureSaturation/CapillaryPressureVanGenuchten.h

```diff
@@ -0,0 +1,147 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * Created on March 20, 2020, 9:59 AM
+ */
+
+#pragma once
+
+#include "SaturationVanGenuchten.h"
+
+namespace MaterialPropertyLib
+{
+ /// \copydoc MaterialPropertyLib::SaturationVanGenuchten
+class CapillaryPressureVanGenuchten : public SaturationVanGenuchten
+{
+public:
+    CapillaryPressureVanGenuchten(double const residual_liquid_saturation,
+                                  double const residual_gas_saturation,
+                                  double const exponent,
+                                  double const entry_pressure,
+                                  double const max_capillary_pressure)
+        : SaturationVanGenuchten(residual_liquid_saturation,
+                                 residual_gas_saturation, exponent,
+                                 entry_pressure),
+          _pc_max(max_capillary_pressure)
+    {
+    }
+
+    void setScale(std::variant<Medium*, Phase*, Component*> scale_pointer)
+    {
+        if (!std::holds_alternative<Medium*>(scale_pointer))
+        {
+            OGS_FATAL(
+                "The property 'CapillaryVanGenuchten' is implemented on the "
+                "'media' scale only.");
+        }
+        this->_medium = std::get<Medium*>(scale_pointer);
+    }
+
+    /// gets \f$ p_c(S) \f$.
+    PropertyDataType value(VariableArray const& variable_array,
+                           ParameterLib::SpatialPosition const& pos,
+                           double const t,
+                           double const dt) const;
```

Comment:
override is missing.

Reply:
I see that you prefer to derive class from Property (in current PR, this one is inherited by RegularizedCapillaryVanGenuchten).